### PR TITLE
New linter. Readme stickers

### PR DESCRIPTION
**Pull request by MattWellie**
# Fixes

  - We have previously had incidents where real sample identifiers have been committed to public-facing repositories

## Proposed Changes

  - Adds the homebrewed CPG pre-commit hook to check for CPGXXX identifiers in project files
  - Adds some additional Ruff rules, e.g. to prevent iSort clashing
  - Adds some vanity stickers to the project root Readme

## Checklist

- [ ] Related Issue created
- [ ] Tests covering new change
- [x] Linting checks pass

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -113,9 +115,9 @@ of all panels on which they have previously been seen. A gene will be treated as
 before, or features in a phenotype-matched panel on which it has not been seen before. i.e.
 
 - if a Gene is promoted to `Green` on the Mendeliome, it will be recorded as `New`, and the prior data will be extended
-to show that the gene was seen on the Mendeliome.
+  to show that the gene was seen on the Mendeliome.
```
Comment (illusional) [inline README.md:118]: IMO, just make this one line, and use soft-wrapping in VSCode:

```jsonc
// in the settings.json
"[markdown]": {
    // ... other settings
    "editor.wordWrap": "wordWrapColumn",
    "editor.wrappingIndent": "same",
    "editor.wordWrapColumn": 80,
},
```

File: helpers/prepare_aip_cohort.py [hunk 2]
```diff
@@ -283,7 +283,7 @@ def main(
     logging.info('Pulling all pedigree members')
     pedigree_dicts, ext_lookup = get_pedigree_for_project(project=project)
 
-    # endpoint gives list of tuples e.g. [['A1234567_proband', 'CPG12341']]
+    # endpoint gives list of tuples e.g. [['A1234567_proband', 'CPT12341']]
```
Comment (illusional) [inline helpers/prepare_aip_cohort.py:286]: You may want to keep the CPG prefix, and make it `CPGaaaa` so it's clear what the second option is here.

**Review by illusional — APPROVED** [review]
Broadly supportive, left some inline comments but you can take or leave

File: README.md (hunk 1 above)
Comment (MattWellie) [reply README.md:118]: This one was caused by markdownlint... or ruff. One or the other. On-save this got split off back onto separate lines. I've reformatted all markdown files in the repo to remove artificial line breaks

**Pull request merged**